### ISSUE TITLE
twin_bottleneck.py

### ISSUE THREAD

**Issue by lpdone** [ymcidence/TBH#1]
hello，Thank you for your excellent work！
`return tf.pow(1 - normalized_dist, 1.4)`
Excuse me, why should the formula for calculating the adjacency matrix be calculated this way? Does 1.4 have a special meaning? Thank you very much.

**Comment by ymcidence** [comment]
> hello，Thank you for your excellent work！
> `return tf.pow(1 - normalized_dist, 1.4)`
> Excuse me, why should the formula for calculating the adjacency matrix be calculated this way? Does 1.4 have a special meaning? Thank you very much.

Hi there. The value of 1.4 itself has no special meaning. It is the distance temperature/sparser for the graph.

This is regarded as a hyper-parameter, which is newly introduced here for future journal submission.

**Comment by fwl2000** [comment]
'feat': tf.io.FixedLenFeature([SET_DIM.get(self.set_name, 4096)], tf.float32),   SET_DIM = 4096, because image feature have 4 channels of RGBA? thanks！

**Comment by ymcidence** [comment]
> 'feat': tf.io.FixedLenFeature([SET_DIM.get(self.set_name, 4096)], tf.float32), SET_DIM = 4096, because image feature have 4 channels of RGBA? thanks！

4096 comes from the length of the CNN feature extractor we use in the experiments. This value is arbitrary as long as you can get your own feature.

**Comment by fwl2000** [comment]
Tanks your reply！
 
From: Yuming Shen
Date: 2020-04-19 13:22
To: ymcidence/TBH
CC: fwl2000; Comment
Subject: Re: [ymcidence/TBH] twin_bottleneck.py (#1)
'feat': tf.io.FixedLenFeature([SET_DIM.get(self.set_name, 4096)], tf.float32), SET_DIM = 4096, because image feature have 4 channels of RGBA? thanks！
4096 comes from the length of the CNN feature extractor we use in the experiments. This value is arbitrary as long as you can get your own feature.

**Comment by lpdone** [comment]
Hello, may I ask you a question? Thank you. Why not test all the test data instead of 400 test images?

**Comment by ymcidence** [comment]
> Hello, may I ask you a question? Thank you. Why not test all the test data instead of 400 test images?

As mentioned in the README, the 'test' routine and figures in the training script are just for illustration. These test figures give us a rough idea of the model performance instead of an accurate score to be reported in the paper. For each hooking step, only 400 test imgs are used to save time. If you need to evaluate the performance accurately, please save the model and run inference on the whole dataset.

**Comment by ymcidence** [comment]
Hi all

I am closing this issue since the main problem is solved.

If you have other questions, please raise another issue.

THX!
